### PR TITLE
SAS-57# add addGamePolicy uc

### PR DESCRIPTION
**Pull request by AmitNachimovitz**

### CODE REVIEW

File: src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        if (rankPolicyId.length() == 0) {
+            return false;
+        }
+
+        LeagueRankPolicy rankPolicy;
+
+        //TODO: check permissions
+        switch (rankPolicyId) {
+            case "1":
+                rankPolicy = new NumberOfWins(league,season);
```
Comment (yaarm) [inline src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java:112]: I think you can can the two rows and put them after the switch case.
league.addRankPolicy(season,rankPolicy);
season.addRankPolicy(league,rankPolicy);

File: src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        gamesPolicies.add(new TwoRoundsLeague());
+        gamesPolicies.add(new ThreeRoundsLeague());
+    }
+
+    /**
+     * The function returns all the available rank policies in our system
+     * @return
+     */
+    public String showRankPolicies(){
+        int counter = 1;
```
Comment (yaarm) [inline src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java:48]: Maybe you can call a function that receives a list and prints the available options - and in each function of show policies will call it with relevant list.

File: src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+            return false;
+        }
+
+        PointsPolicy pointsPolicy;
+
+        //TODO: check permissions
+        switch (pointsPolicyId) {
+            case "1":
+                pointsPolicy = new OnePointForWinAndNoneForDraw(league,season);
+                league.addPointsPolicy(season,pointsPolicy);
```
Comment (yaarm) [inline src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java:154]: The same here about the settings.

**Review by MrBahur — APPROVED** [review]
looking good, merging

File: src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+    /**
+     * The function receives league, season and game policy id and set the game policy to the league according to the season
+     * @param league
+     * @param season
+     * @param gamePolicyId
+     * @return
+     */
+    public boolean addGamePolicy(League league, Season season, String gamePolicyId){
+        if (league == null || season == null) {
```
Comment (yaarm) [inline src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java:187]: Maybe add a function of validate in each function.

Comment (MrBahur) [reply src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java:187]: concider doing it in feauture task, right now its ok

File: src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java (hunk 2 above)
Comment (MrBahur) [reply src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java:48]: concider doing it in feauture task, right now its ok

File: src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java (hunk 1 above)
Comment (MrBahur) [reply src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java:112]: agree

**Pull request merged**